backend/attendance-service/src/main/java/org/egov/repository/AttendanceRepository.java

```diff
@@ -36,32 +36,47 @@ public class AttendanceRepository {
 
     public List<AttendanceRegister> getAttendanceRegister(AttendanceRegisterSearchCriteria searchCriteria) {
         List<Object> preparedStmtList = new ArrayList<>();
-        List<String> registerIds = new ArrayList<>();
+        List<String> registerIds;
+        List<AttendanceRegister> attendeeRegisters = null;
 
-        String attendeeQuery = attendeeQueryBuilder.getAttendeeSearchQuery(searchCriteria, preparedStmtList);
-        List<AttendanceRegister> attendanceRegisters = jdbcTemplate.query(attendeeQuery, attendeeRowMapper, preparedStmtList.toArray());
+        if (StringUtils.isBlank(searchCriteria.getStaffId())) {
+            String attendeeQuery = attendeeQueryBuilder.getAttendeeSearchQuery(searchCriteria, preparedStmtList);
+            attendeeRegisters = jdbcTemplate.query(attendeeQuery, attendeeRowMapper, preparedStmtList.toArray());
 
-        //If the search is based on attendeeId , fetch the matched attendance registerIds to pass it to staffQueryBuilder
-        if (StringUtils.isNotBlank(searchCriteria.getAttendeeId())) {
-            registerIds = attendanceRegisters.stream()
-                    .map(register -> register.getId().toString()).collect(Collectors.toList());
+            //If the search is based on attendeeId , fetch the matched attendance registerIds to pass it to staffQueryBuilder
+            if (StringUtils.isNotBlank(searchCriteria.getAttendeeId())) {
```

Comment:
This should be doable without nested if statements. Please try to implement that approach.

Reply:
Refactored the code by removing the nested if statements.